tests/rptest/tests/materialize_test.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from ducktape.mark.resource import cluster
+from ducktape.mark import matrix
+from ducktape.tests.test import Test
+from ducktape.utils.util import wait_until
+from re import match
+from rptest.services.redpanda import RedpandaService
+
+import subprocess
+
+
+class MaterializeIter:
+    def __init__(self, lines):
+        self.lines = lines
+        self.index = -1
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        self.index += 1
+        if self.index < len(self.lines):
+            return self.lines[self.index]
+
+        raise StopIteration
+
+
+class MaterializeTest(Test):
+    MATERIALZE_HOME = '/opt/materialize'
+    SERVICE_FILE_PY = f'{MATERIALZE_HOME}/misc/python/materialize/mzcompose/services.py'
+    TESTDRIVE_DIR = f'{MATERIALZE_HOME}/test/testdrive'
+
+    # These are the versions of the test suite that each
+    # minor release of Redpanda should be tested against.
+    VERSION_COMMIT_MAP = {
+        # RP minor release --> SHA @ github.com/MaterializeInc/materialize
+        'v21.11.13': '2f0e02098aad303f25b758cb922f5134fb1d781c',
+        'v22.1.3': 'a6ef2866462b9c34c53c9f60f413353f69ae8d2d',
+        'v22.1.4': 'a6ef2866462b9c34c53c9f60f413353f69ae8d2d',
+        'v22.1.5-rc1': 'a6ef2866462b9c34c53c9f60f413353f69ae8d2d'
+    }
+
+    def __init__(self, test_context):
+        super(MaterializeTest, self).__init__(test_context=test_context)
+        # Capturing dedicated nodes manually instead of via RedpandaService
+        # because the materialize test suite already sets up a RP cluster.
+        # So there is no need to startup our own RP cluster.
+        self.dedicated_nodes = self.test_context.globals.get(
+            RedpandaService.DEDICATED_NODE_KEY, False)
+
+    '''
+    Typically we avoid running auxillary apps on the client node.
+    The materialize tests, however, use more disk space than our worker
+    CDT nodes have available. The tests successfully run on the CDT client
+    node.
+
+    The easy fix would be to use larger instance types on the workers but the
+    materialize tests require a single node only because they setup a single RP
+    broker within a container. So increasing the instance type for *all* worker
+    nodes is wasteful.
+
+
+    Instead let's use the client node for materialize tests and ensure
+    the cleanup process reverts all changes.
+    '''
```

Comment:
Will discuss this in standup/core-panda

Reply:
The team decided that we should stick to our testing strategy: "do not run anything on the client node"

So we will need to figure out a way to either:
1. reduce the size of materialize tests (currently requires ~11G of space)
2. get more disk space for a subset of CDT nodes

Reply:
> reduce the size of materialize tests (currently requires ~11G of space)

this seems like quite a small of small. perhaps we should expect nodes to have additional space available. do you know which device is reaching capacity? presumably it is the root device, which should be easy enough to increase in capacity.

Reply:
This is actually an OOM error instead of disk space. The host system is running out of memory.